src/main/java/com/sonymobile/tools/gerrit/gerritevents/GerritConnection.java

```diff
@@ -94,6 +94,7 @@
     private AuthenticationUpdater authenticationUpdater = null;
     private final Set<ConnectionListener> listeners = new CopyOnWriteArraySet<ConnectionListener>();
     private int sshRxBufferSize = SSH_RX_BUFFER_SIZE;
+    private StringBuilder eventBuffer = null;
```

Comment:
> StringBuilder is not thread safe, but it's perfectly fine for this case

Really???

```
GerritConnection con = new GerritConnection(...)
for (int i = 0; i < 3; i++) {
  Thread thread = new Thread(con)
  thread.run()
}
````
Thread implements Runnable. So the instance of GerritConnection can be used by multiple threads. (Though nobody want to use GerritConnection instance with such way)